lib/rspec/core/notifications.rb

```diff
@@ -577,23 +577,19 @@ def percentage
       end
 
       # @return [Array<RSpec::Core::Example>] the slowest example groups
-      def slowest_groups
-        @slowest_groups ||= calculate_slowest_groups
-      end
-
-    private
-
-      def calculate_slowest_groups
+      def calculate_slowest_groups(execution_times)
         example_groups = {}
+        slowest_first = execution_times.sort.reverse.first(number_of_examples).to_h
 
         examples.each do |example|
           location = example.example_group.parent_groups.last.metadata[:location]
-
-          location_hash = example_groups[location] ||= Hash.new(0)
-          location_hash[:total_time]  += example.execution_result.run_time
-          location_hash[:count]       += 1
-          next if location_hash.key?(:description)
-          location_hash[:description] = example.example_group.top_level_description
+          if(slowest_first.has_key?(location))
+            location_hash = example_groups[location] ||= Hash.new(0)
+            location_hash[:count]       += 1
+            next if location_hash.key?(:description)
+            location_hash[:description] = example.example_group.top_level_description
+            location_hash[:total_time] = slowest_first[location]
```

Comment:
My reading of this is you're now ignoring the example run time which is probably correct, but there should be no need to loop over all the examples if you have all the group run times already?